### ISSUE TITLE
Phantom Peaks in FEV

### ISSUE THREAD

**Issue by HansVraga** [USGS-WiM/STNWeb#435]
We've got some phantom peaks in FEV!  Shelby in Iowa is having problems in at least three events listed in this email thread.  I investigated one...

If you select the Iowa June 2008 Boyer River Flood, eight sites show up in STN:

![image](https://user-images.githubusercontent.com/12058013/42115740-39cec2a8-7bc2-11e8-8d32-c20b03715b9a.png)

(Two of them {26181 and 26180} are probably too close to each other to be separate sites, but that may be unrelated.)
If you select the same event in FEV, you see the same eight sites represented mostly by HWMs and peaks, but there is a ninth site represented only by a peak that is a few counties away to the northwest.
![image](https://user-images.githubusercontent.com/12058013/42115786-68549242-7bc2-11e8-9f24-d5a1917358a1.png)

Clicking on that phantom peak isn't very useful because we can't see any site information from the peak itself -- something we ought to fix in future versions.
At this point, there's no easy way to find that phantom peak in STN because it isn't showing up on the list or the map when the Boyer event is selected.
However, you can go to GET DATA in FEV and download the Peak Summary, and the phantom peak DOES show up there (albeit, with a different river name):

![image](https://user-images.githubusercontent.com/12058013/42115791-7142755e-7bc2-11e8-9905-5df0490761b5.png)

If we lookup this site number in STN, it shows no peaks for the 2008 Boyer event.  In fact, it shows no peaks at all, even when no events are selected, even though it counts one in the accordion bar:

![image](https://user-images.githubusercontent.com/12058013/42115799-7a629e70-7bc2-11e8-9c3d-ae9dd8d320b0.png)

Also interesting to note is that IACRA26176 does not show up in the CSV export, even though that site DOES show up in the STN list for this event:

![image](https://user-images.githubusercontent.com/12058013/42115806-8547ffc4-7bc2-11e8-9ce5-c6171b5487c7.png)

So to sum up, in the Iowa June 2008 Boyer River Flood event, we have...
a peak (site IAWRI26010) that shows up in FEV but doesn't show up as a Site in STN, and 
a site (IACRA26176) that shows up in STN for the 2008 Boyer event but does not show up in FEV.
I suspect there are similar circumstances for the other two events that Shelby mentioned, but I haven't had the time to fully investigate those.  Can you guys take it from here?

Thanks,
-Todd

**Comment by lprivette** [comment]
When adding a peak_summary to a HWM we noticed that if the fields entered (peak_date, for example) were the same as the previous peak entered, that the services would see that as being already in the DB. Because of that, that peak would be duplicated and a new one would not be created. So after create a peak, if you go to a new site and start creating a new peak, but the date/time doesn't change, it would then use the previously entered peak summary for that different site/HWM. And so, like with the example peak (10176) that I was showing you Todd, that peak had been assigned to 5 different sites and HWMs in the "Iowa July 1993 Flood Des Moines River" event.

To fix it, Jeremy duplicated what was there into new peaks, then updated the HWM with the new peak summary ID

Fixed in PROD

https://github.com/USGS-WiM/STNServices2/commit/f4ecf9d3dd15e87bcacee5b498a3aaaee0855ebc
